elements/lisk-codec/src/codec.ts

```diff
@@ -108,7 +108,7 @@ export class Codec {
 			this.addSchema(schema);
 		}
 		const compiledSchema = this._compileSchemas[schema.$id];
-		const [res] = readObject(message, 0, compiledSchema);
+		const [res] = readObject(message, 0, compiledSchema, message.length);
```

Comment:
Would not it be always `message.lenth` if `terminateIndex` is not provided? What other cases there could be to not provide  number of bytes to read from the message and you also don't want to read the full message. 

May be I am missing some context of the codec here, but it seems if we moved this logic inside the `readObject` that if `terminateIndex` is not provided it would always be `message.length - offset` that will a safe option.

Reply:
root level terminating index is always the size of the message, but each internal object property will have it's own terminating length.
Also, this is the entry point of the recursive function

Reply:
Yes, I understand that and that's why I suggested if `terminateIndex` is not provided, for internal object properties we already have terminating length provided.